components/sceneutil/shadow.hpp

```diff
@@ -19,15 +19,13 @@ namespace SceneUtil
 
         ShadowManager(osg::ref_ptr<osg::Group> sceneRoot, osg::ref_ptr<osg::Group> rootNode, unsigned int outdoorShadowCastingMask, unsigned int indoorShadowCastingMask, Shader::ShaderManager &shaderManager);
 
-        virtual ~ShadowManager() = default;
+        void setupShadowSettings();
 
-        virtual void setupShadowSettings();
+        Shader::ShaderManager::DefineMap getShadowDefines();
 
-        virtual Shader::ShaderManager::DefineMap getShadowDefines();
+        void enableIndoorMode();
 
-        virtual void enableIndoorMode();
-
-        virtual void enableOutdoorMode();
+        void enableOutdoorMode();
```

Comment:
This section of the diff is a mess which will make git blame harder to use. Is there some way you could regenerate it so that it's more obvious which lines came from where?

Also, the whole things-being-virtual approach was so that we could potentially have other shadow implementations as a drop-in replacement (e.g. the one MP3Butcher was working on and then abandoned), but I guess if that ever happens, we can just make them virtual again.

Reply:
Not sure what do you mean. Blame looks fine: 
<img width="751" alt="blame" src="https://user-images.githubusercontent.com/764107/69320341-51065b00-0c41-11ea-9e0c-1137d7d20063.png">
Diff itself is not very good but it's because github interface does it. Git [doesn't store diffs](https://stackoverflow.com/questions/10398744/does-git-store-diff-information-in-commit-objects/10398894#10398894), it stores full files content. So how to generate diff is not in my power here.

About virtual I think it's better to add it when it's required. Also more recommended approach is to create class with pure virtual methods to define interface.